tests/client/core/TestAddonManager.js

```diff
@@ -91,12 +91,20 @@ describe('addonManager', () => {
     it('passes the installObj, the event and the id to the callback', () => {
       const fakeEvent = { type: 'fakeEvent' };
       return addonManager.install(
-        fakeInstallUrl, fakeCallback, { _mozAddonManager: fakeMozAddonManager })
+        fakeInstallUrl, fakeCallback, { _mozAddonManager: fakeMozAddonManager, src: 'home' })
         .then(() => {
           fakeInstallObj.onDownloadProgressListener(fakeEvent);
           assert.ok(fakeCallback.calledWith(fakeInstallObj, fakeEvent));
         });
     });
+
+    it('requires a src', () => (
+      addonManager.install(
+        fakeInstallUrl, fakeCallback, { _mozAddonManager: fakeMozAddonManager })
+          .then(
```

Comment:
If the unexpected success assertion throws in the `then` we want the test runner to catch it so we get the "unexpected success" message. 

If there was a catch as you suggest you'd only see the exception was wrong which is makes the reason for the failure less clear.

See http://jsbin.com/yatofoc/edit?js,console for an example of the differences.